libsql-server/src/query_analysis.rs

```diff
@@ -332,15 +332,41 @@ impl Statement {
         // on the heap:
         // - https://github.com/gwenn/lemon-rs/issues/8
         // - https://github.com/gwenn/lemon-rs/pull/19
-        let mut parser = Box::new(Parser::new(s.as_bytes()).peekable());
+        let mut parser = Some(Box::new(Parser::new(s.as_bytes()).peekable()));
         let mut stmt_count = 0;
         std::iter::from_fn(move || {
+            // temporary macro to catch panic from the parser, until we fix it.
+            macro_rules! parse {
```

Comment:
do we really need a match for two uses 😂

Reply:
it gets very clumsy, ideally we get rid of that asap